include/__generator.hpp

```diff
@@ -451,18 +464,20 @@ struct __generator_promise_base
         return std::move(__g).get();
     }
 
-    template <std::ranges::range _Rng>
-    __yield_sequence_awaiter<generator<_Ref>> yield_value(std::ranges::elements_of<_Rng> __x) {
+    template <std::ranges::range _Rng, typename _Allocator>
+    __yield_sequence_awaiter<generator<_Ref, std::ranges::range_value_t<_Rng>, _Allocator>>
+    __yield_value(std::ranges::elements_of<_Rng, _Allocator> && __x) {
         // TODO: Consider propagating parent coroutine's allocator to child generator
         // coroutine here.
-        return yield_value(std::ranges::elements_of([](_Rng&& __rng) -> generator<_Ref> {
+        // TODO: Should the promise type be templated on value to reduce template instantiations?
+        return [](auto && __rng, allocator_arg_t, _Allocator alloc) -> generator<_Ref, std::ranges::range_value_t<_Rng>, _Allocator> {
```

Comment:
Note that allocator_arg_t usually needs to be the first arg to be able to trigger allocator customisation.
I think this is working here because we also need to support allocator_arg in the second position when used as a member function as the first arg is taken up by the implicit object parameter.

Should we move allocator_arg_t to the first arg here to be more obviously correct?

Reply:
Indeed, thanks